presto-main/src/main/java/com/facebook/presto/operator/scalar/RegexpFunctions.java

```diff
@@ -15,300 +15,106 @@
 
 import com.facebook.presto.metadata.OperatorType;
 import com.facebook.presto.operator.Description;
+import com.facebook.presto.spi.ConnectorSession;
 import com.facebook.presto.spi.PrestoException;
 import com.facebook.presto.spi.block.Block;
-import com.facebook.presto.spi.block.BlockBuilder;
-import com.facebook.presto.spi.block.BlockBuilderStatus;
 import com.facebook.presto.spi.type.StandardTypes;
 import com.facebook.presto.type.RegexpType;
 import com.facebook.presto.type.SqlType;
-import com.google.common.primitives.Ints;
-import io.airlift.jcodings.specific.NonStrictUTF8Encoding;
-import io.airlift.joni.Matcher;
-import io.airlift.joni.Option;
-import io.airlift.joni.Regex;
-import io.airlift.joni.Region;
-import io.airlift.joni.Syntax;
-import io.airlift.joni.exception.ValueException;
-import io.airlift.slice.DynamicSliceOutput;
+
 import io.airlift.slice.Slice;
-import io.airlift.slice.SliceOutput;
 import io.airlift.slice.Slices;
 
 import javax.annotation.Nullable;
 
-import java.nio.charset.StandardCharsets;
-
 import static com.facebook.presto.spi.StandardErrorCode.INVALID_FUNCTION_ARGUMENT;
-import static com.facebook.presto.spi.type.VarcharType.VARCHAR;
-import static java.lang.String.format;
 
 public final class RegexpFunctions
 {
-    private RegexpFunctions()
-    {
-    }
+    private RegexpFunctions(){}
 
     @ScalarOperator(OperatorType.CAST)
     @SqlType(RegexpType.NAME)
-    public static Regex castToRegexp(@SqlType(StandardTypes.VARCHAR) Slice pattern)
+    public static RegexpGenericPattern castToRegexp(ConnectorSession session, @SqlType(StandardTypes.VARCHAR) Slice pattern)
     {
-        Regex regex;
+        String regexLibraryName = session.getSystemProperty("regex_library", String.class);
+        if (regexLibraryName == null) {
+            regexLibraryName = "JONI"; //default is JONI
```

Comment:
default should be extracted automatically when getSystemProperty is implemented with using `sessionPropertyManager`